internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -1216,10 +1224,157 @@ func (c *Coordinator) generateComponentModel() (err error) {
 	// If we made it this far, update our internal derived values and
 	// return with no error
 	c.derivedConfig = cfg
+
+	lastComponentModel := c.componentModel
 	c.componentModel = comps
+
+	// don't run a bunch of relatively costly code unless we're in debug mode
+	if c.logger.IsDebug() {
+		c.checkAndLogUpdate(lastComponentModel)
+	}
```

Comment:
Agreed this should be at the info level. Policy changes producing differences are rare but important. Also the resource cost of this will be dominated by starting and stopping subprocesses.